nomos-services/data-availability/dispersal/src/adapters/mempool/kzgrs.rs

```diff
@@ -0,0 +1,115 @@
+// std
+use std::hash::Hash;
+use std::{fmt::Debug, marker::PhantomData};
+// crates
+use kzgrs_backend::dispersal::{self, BlobInfo};
+use nomos_core::da::blob::info::DispersedBlobInfo;
+use nomos_core::header::HeaderId;
+use nomos_da_sampling::backend::DaSamplingServiceBackend;
+use nomos_mempool::{
+    backend::MemPool, network::NetworkAdapter as MempoolAdapter, DaMempoolService, MempoolMsg,
+};
+use overwatch_rs::{
+    services::{relay::OutboundRelay, ServiceData},
+    DynError,
+};
```

Comment:
nitpick: this ones would be internal (although they are crates they are internal to us).